server/server.js

```diff
@@ -10,36 +14,33 @@ server.use(jsonServer.rewriter({
     "/api/documents/popular": "/documents?_sort=popular&_order=desc&_limit=8",
     "/api/documents?search=:search": "/documents?q=:search",
     "/api/documents?search=:search&filterId=:filter": "/documents?q=:search&filterId=:filter",
-    "/api/documents?search=:search&filtersCount": "/documents?q=:search",
     "/api/*": "/$1"
 }));
-server.use(router);
 
-router.render = function (req, res) {
-    if (req._parsedOriginalUrl.query && req._parsedOriginalUrl.query.includes('filtersCount')) {
-        const filters = [];
-        const cache = {};
-        res.locals.data.forEach((elem) => {
-            const {filterId} = elem;
-            if (cache[filterId]) {
-                cache[filterId] += 1;
-            } else {
-                cache[filterId] = 1;
-            }
-        });
-        Object.keys(cache).forEach((key) => {
-            filters.push({
-                filterId: key,
-                filtersQuantity: cache[key],
-            });
-        });
-        res.jsonp({
-            filters: filters
-        })
-    } else {
-        res.jsonp(res.locals.data)
+
+server.use((req, res, next) => {
+    const { documents } = data;
+    const { q: searchQuery, filtersCount } = req.query;
+    if (!filtersCount) {
+        return next();
```

Comment:
http://localhost:3000/api/documents?search=taxes&filtersCount=false, should return nothing. Check, please.

Reply:
Fixed